internal/actions/drain_node_handler_test.go

```diff
@@ -509,29 +508,21 @@ func TestLogCastPodsToEvict(t *testing.T) {
 	})
 }
 
-func prependEvictionReaction(t *testing.T, c *fake.Clientset, success, retryableFailure bool) {
-	c.PrependReactor("create", "pods", func(action ktest.Action) (handled bool, ret runtime.Object, err error) {
-		if action.GetSubresource() != "eviction" {
-			return false, nil, nil
-		}
-
-		if success {
-			eviction := action.(ktest.CreateAction).GetObject().(*v1beta1.Eviction)
-
-			go func() {
-				err := c.CoreV1().Pods(eviction.Namespace).Delete(context.Background(), eviction.Name, metav1.DeleteOptions{})
-				require.NoError(t, err)
-			}()
-
-			return true, nil, nil
-		}
-
-		// Simulate failure that should be retried by client.
-		if retryableFailure {
-			return true, nil, &apierrors.StatusError{ErrStatus: metav1.Status{Reason: metav1.StatusReasonTooManyRequests}}
+func prependEvictionReaction(t testing.TB, c *fake.Clientset, success, retryableFailure bool) {
+	prependPodEvictionReaction(c, func(namespace, name string) error {
+		if !success {
+			if retryableFailure {
+				// Simulate failure that should be retried by client.
+				return &apierrors.StatusError{ErrStatus: metav1.Status{Reason: metav1.StatusReasonTooManyRequests}}
+			} else {
```

Comment:
NITPICK: I suggest to get rid of this `else` (generally speaking they are ugly ;)